app/components/taskLogs/taskLogsController.js

```diff
@@ -0,0 +1,77 @@
+angular.module('taskLogs', [])
+.controller('TaskLogsController', ['$scope', '$stateParams', 'ServiceService', 'TaskService', 'Notifications',
+function ($scope, $stateParams, ServiceService, TaskService, Notifications) {
+  $scope.state = {};
+  $scope.state.displayTimestampsOut = false;
+  $scope.state.displayTimestampsErr = false;
+  $scope.stdout = '';
+  $scope.stderr = '';
+  $scope.tailLines = 2000;
+
+  function getLogs() {
+    $('#loadingViewSpinner').show();
+    getLogsStdout();
+    getLogsStderr();
+    $('#loadingViewSpinner').hide();
+  }
+
+  function parseLogResults(data) {
+    // Replace carriage returns with newlines to clean up output
+    data = data.replace(/[\r]/g, '\n');
+    // Strip 8 byte header from each line of output
+    data = data.substring(8);
+    data = data.replace(/\n(.{8})/g, '\n');
+    return data;
+  }
```

Comment:
This is the same across all controllers, it could be moved to the separate service

Reply:
Agreed. All log controllers are now pretty much the same, so we can easily move most of it to a separate service.